internal/service/federatedsettingsidentityprovider/data_source_federated_settings_identity_providers_test.go

```diff
@@ -20,22 +22,66 @@ func TestAccFederatedSettingsIdentityProvidersDS_basic(t *testing.T) {
 		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
 		Steps: []resource.TestStep{
 			{
-				Config: configBasicPluralDS(federatedSettingsID),
+				Config: configPluralDS(federatedSettingsID, nil, []string{oidcProtocol, samlProtocol}),
 				Check: resource.ComposeTestCheckFunc(
 					resource.TestCheckResourceAttrSet(dataSourceName, "federation_settings_id"),
 					resource.TestCheckResourceAttr(dataSourceName, "results.#", "2"),
 				),
 			},
+			{
+				Config: configPluralDS(federatedSettingsID, nil, []string{samlProtocol}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "federation_settings_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.0.display_name", "SAML-test"),
+				),
+			},
+			{
+				Config: configPluralDS(federatedSettingsID, nil, []string{oidcProtocol}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "federation_settings_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.0.display_name", "OIDC-test"),
+				),
+			},
+			{
+				Config: configPluralDS(federatedSettingsID, nil, []string{}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "federation_settings_id"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.#", "1"),
+					resource.TestCheckResourceAttr(dataSourceName, "results.0.display_name", "SAML-test"), // if no protocol is specified, it defaults to SAML
+				),
+			},
+			{
+				Config: configPluralDS(federatedSettingsID, conversion.StringPtr(federatedsettingsidentityprovider.WORKLOAD), []string{}),
```

Comment:
[q] are we covering `WORKFORCE`too?

Reply:
If not set, it defaults to `WORKFORCE`, so in the other tests cases we are testing it. I understand that it might not be super clear seeing the test. I will change it so it's explicit